__tests__/__unit__/DatasetTree.unit.test.ts

```diff
@@ -171,8 +176,14 @@ describe("DatasetTree Unit Tests", () => {
         sessNode[0].contextValue = extension.FAVORITE_CONTEXT;
         sessNode[1].contextValue = extension.DS_SESSION_CONTEXT;
         sessNode[1].pattern = "test";
-        sessNode[0].iconPath = utils.applyIcons(sessNode[0]);
-        sessNode[1].iconPath = utils.applyIcons(sessNode[1]);
+        let targetIcon = getIconByNode(sessNode[0]);
+        if (targetIcon) {
+            sessNode[0].iconPath = targetIcon.path;
+        }
+        targetIcon = getIconByNode(sessNode[1]);
+        if (targetIcon) {
+            sessNode[1].iconPath = targetIcon.path;
+        }
```

Comment:
I feel like having a TestEnvironment will be helpful for handling this kinds of things in the future.
It may help to reduce not only volume of test files but also complexity. :smile:

No changes requested!